File: issta2018artifacts/minimized-bugs/buggy-key/Duplicates.java
// Test case for plume-lib bug fixed in 8224707


public class Duplicates {

	/*@ public normal_behavior
@ requires nums.length >= 1;
@ ensures true;
@*/
    public static int[] missing_numbers(int [] nums) {
	{ // avoid modifying parameter
	    int[] nums_copy = new int[nums.length];
	    nums = nums_copy;
	}
	int min = nums[0];
	int max = nums[nums.length-1];
	//:: error: (array.length.negative)
	int[] result = new int[max - min + 1 - nums.length];
	return result;
     }
}

File: issta2018artifacts/minimized-bugs/buggy-clousot/PlumeLib/Duplicates.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.Contracts;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


public class Duplicates
{
    public static int[] missing_numbers(int[] nums) {
        Contract.Requires(nums.Length >= 1);
	    { // avoid modifying parameter
	        int[] nums_copy = new int[nums.Length];
            Array.Copy(nums, 0, nums_copy, 0, nums.Length);
	        nums = nums_copy;
	    }
        Array.Sort(nums);
	    int min = nums[0];
        int max = nums[nums.Length - 1];
        //:: error: (array.length.negative)
        int[] result = new int[max - min + 1 - nums.Length];
	    return result;
    }
}